### DISCUSSION TITLE
Per upstream object configurable keepalive timeout option

### DISCUSSION THREAD

**Discussion by zeghong** [Kong/kong#8302] [Ideas and feature requests]
I was wondering if we can add an option to customize the `upstream_keepalive_idle_timeout`  per upstream object.

There is a global configuration `upstream_keepalive_idle_timeout` ([Link](https://docs.konghq.com/gateway/2.7.x/reference/configuration/#upstream_keepalive_idle_timeout)) which controls the global default upstream keepalive timeout, but i think it would be helpful if we can customize this value per upstream object.

Here is a typical scenario:
*Gunicorn* is a popular Python WSGI HTTP server, and it's basic [sync worker](https://docs.gunicorn.org/en/stable/design.html#sync-workers) does not support persistent connections. So the [deployment guide](https://docs.gunicorn.org/en/stable/deploy.html#nginx-configuration) recommends the nginx directive `keepliave_timeout` been set to `5` (seconds), which is shorter than kong's default value `60`。It the keepalive idle timeout can be customized without affecting other workload with persistent connections support, it will helps to reduce the number of `TIME_WAIT` TCP sockets.

Since the original nginx `keepalive_timeout` [directive](http://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive_timeout) value is set in the *upstream* context, i assumes this feature is realizable from technical point of view.

**Comment by guoard** [comment]
+1
